apps/backend/src/lib/controller/chip/managed/leaderboard/index.ts

```diff
@@ -63,127 +82,140 @@ ManagedChipClient.prototype.UpdateLeaderboardEntry = async function (
-      length = totalTaps.length;
-    }
-    throw new Error(
-      `Failed to get total taps of leaderboard entries, return value of length ${length}`
-    );
-  } catch (error) {
-    console.error(
-      "Failed to get total taps of leaderboard entries:",
-      errorToString(error)
-    );
-    throw new Error("Failed to get total taps of leaderboard entries");
+  // TODO: This is a temporary solution to handle the lanna total tap count.
+  // In the future, we will migrate this to the general leaderboard entry update.
+  if (
+    chipIssuer === ChipIssuer.EDGE_CITY_LANNA &&
+    entryType === LeaderboardEntryType.TOTAL_TAP_COUNT
+  ) {
+    const totalValue = await this.prismaClient.leaderboardEntry.aggregate({
+      _sum: {
+        tapCount: true,
+      },
+      where: {
+        chipIssuer,
+        entryType: null,
+      },
+    });
+
+    return totalValue._sum.tapCount ?? null;
   }
 
-  return null;
+  const totalValue = await this.prismaClient.leaderboardEntry.aggregate({
+    _sum: {
+      entryValue: true,
+    },
+    where: {
+      chipIssuer,
+      entryType,
+    },
+  });
+
+  return totalValue._sum.entryValue?.toNumber() ?? null;
 };
 
 // Get contributors from chip service rather than postgres.
 // While postgres handles user details, the chip service builds up a record of its community members through entries.
 // LeaderboardEntry count can represent the number of contributors.
 ManagedChipClient.prototype.GetLeaderboardTotalContributors = async function (
-  chipIssuer: ChipIssuer
+  chipIssuer: ChipIssuer,
+  entryType: LeaderboardEntryType
 ): Promise<number | null> {
-  try {
-    // There should be one distinct entry per user-leaderboard
-    const totalContributors: Array<{ count: number } | null> = await this
-      .prismaClient
-      .$queryRaw`SELECT COUNT(*) FROM "LeaderboardEntry" WHERE "chipIssuer"=${chipIssuer}`;
-
-    let length = -1;
-    if (
-      totalContributors &&
-      totalContributors.length == 1 &&
-      totalContributors[0] !== null
-    ) {
-      // List of total contributors should always be length 1
-      return Number(totalContributors[0].count);
-    } else if (totalContributors) {
-      // in case the query returns an unexpected result
-      length = totalContributors.length;
-    }
-    throw new Error(
-      `Failed to get total taps of leaderboard entries, return value of length ${length}`
-    );
-  } catch (error) {
-    console.error(
-      "Failed to get total contributor count of leaderboard:",
-      errorToString(error)
-    );
-    throw new Error("Failed to get total contributor count of leaderboard");
+  // TODO: This is a temporary solution to handle the lanna total tap count.
+  // In the future, we will migrate this to the general leaderboard entry update.
+  if (
+    chipIssuer === ChipIssuer.EDGE_CITY_LANNA &&
+    entryType === LeaderboardEntryType.TOTAL_TAP_COUNT
+  ) {
+    const totalContributors = await this.prismaClient.leaderboardEntry.count({
```

Comment:
Not important -- but could have sworn I checked for a count operator. Thanks for swapping to using prisma.